cmsplugin_feed_ai/feed.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+from warnings import warn
+
+from django.conf import settings
+from django.core.cache import cache
+from facebook import GraphAPI
+from twython import Twython
+
+from .providers.facebook import FacebookFeedDisplayItem
+from .providers.twitter import TwitterFeedDisplayItem
+
+DEFAULT_CACHE_TIME = 3600  # 1 hour
+
+SERVICE_TO_DISPLAY_ITEM_CLASS_MAP = {
+    "facebook": FacebookFeedDisplayItem,
+    "twitter": TwitterFeedDisplayItem,
+}
+
+SUPPORTED_SERVICES = set(SERVICE_TO_DISPLAY_ITEM_CLASS_MAP.keys())
+
+
+def get_feed_settings(key, default=None):
+    feed_settings = getattr(settings, "AI_FEED_SETTINGS", {})
+    if key not in feed_settings:
+        warn("Missing %r setting in AI_FEED_SETTINGS" % key)
+        return default
+    return feed_settings.get(key, default)
+
+
+def _fetch_twitter_profile_feed(profile_name, limit):
```

Comment:
This should live in `.providers.twitter` as well, and should likewise be mapped via a dict:

```python
SERVICE_TO_FETCHER_MAP = {
    "facebook": ....,
    "twitter": fetch_twitter_profile_feed,
}
```